crates/ark/src/version.rs

```diff
@@ -30,19 +30,35 @@ pub struct RVersion {
     pub r_home: String,
 }
 
-pub fn detect_r() -> anyhow::Result<RVersion> {
-    let output = r_command(|command| {
-        command.arg("RHOME");
-    })
-    .context("Failed to execute R to determine R_HOME")?;
+pub(crate) fn r_home_setup() -> PathBuf {
+    match std::env::var("R_HOME") {
+        Ok(home) => {
+            // Get `R_HOME` from env var, typically set by Positron / CI / kernel specification
+            PathBuf::from(home)
+        },
+        Err(_) => {
+            // Get `R_HOME` from `PATH`, via `R`
+            let Ok(result) = r_command_from_path(|command| {
+                command.arg("RHOME");
+            }) else {
+                panic!("Can't find R or `R_HOME`");
+            };
+
+            let r_home = String::from_utf8(result.stdout).unwrap();
+            let r_home = r_home.trim();
+
+            // Now set `R_HOME`. From now on, `r_command()` can be used to
+            // run exactly the same R as is running in Ark.
+            unsafe { std::env::set_var("R_HOME", r_home) };
+            PathBuf::from(r_home)
+        },
+    }
+}
 
-    // Convert the output to a string
-    let r_home = String::from_utf8(output.stdout)
-        .context("Failed to convert R_HOME output to string")?
-        .trim()
-        .to_string();
+pub fn detect_r() -> anyhow::Result<RVersion> {
+    let r_home: String = r_home_setup().to_string_lossy().to_string();
 
-    let output = r_command(|command| {
+    let output = r_command_from_path(|command| {
```

Comment:
I think it is best to call `r_command()` here because you called `r_home_setup()` right above this, which sets you up to be able to call `r_command()` with the same `R_HOME` that `r_home_setup()` found.